### ISSUE TITLE
--severity low cannot override a config severity, because the default doubles as the sentinel

### ISSUE THREAD

**Issue by CognitiveDecisions** [rsionnach/sloppylint#27]
`merge_cli_args` decides whether `--severity` was passed by comparing it to its own default:

https://github.com/rsionnach/sloppylint/blob/e0e3ced/src/sloppy/config.py#L77

```python
if hasattr(args, "severity") and args.severity != "low":
    self.severity = args.severity
```

`--severity low` is therefore indistinguishable from not passing the flag, so it cannot lower a threshold set in config.

### Repro

```
$ printf '[tool.sloppy]\nseverity = "high"\n' > pyproject.toml
$ sloppylint s.py --severity low --format compact | grep -c overconfident
0                                    # medium finding suppressed

$ cd .. && sloppylint p3/s.py --severity low --format compact | grep -c overconfident
1                                    # control: same file, no config, flag works
```

### Suggested fix

Use `None` as the argparse default and treat `None` as absent:

```python
parser.add_argument("--severity", "-s", choices=[...], default=None, ...)
```

```python
if getattr(args, "severity", None) is not None:
    self.severity = args.severity
```

The same shape appears just below for `--format`, which compares against `"detailed"`, so `--format detailed` cannot override a config `format` either.

Found on 0.5.1 (`e0e3ced`).
